GPUTest.test_numpy_pad_modes_match_scipy_median_modes failure

### Summary

test_numpy_pad_modes_match_scipy_median_modes() fails on my laptop. (but passes on my VM)

### Steps To Reproduce

`python -m pytest  -s -k GPU`

### Expected Behaviour

Pass

### Current Behaviour

```
_________________________________________________________________________________ GPUTest.test_gpu_result_matches_cpu_result_for_larger_images ___________________________________________________________________________________

self = <mantidimaging.core.gpu.test.gpu_test.GPUTest testMethod=test_gpu_result_matches_cpu_result_for_larger_images>

    @unittest.skipIf(GPU_NOT_AVAIL, reason=GPU_SKIP_REASON)
    def test_gpu_result_matches_cpu_result_for_larger_images(self):
        """
        Run the median filter on the CPU and GPU with a larger image size. Check that the results match. This test may
        reveal issues such as the grid and dimension size arguments going wrong.
        """
        N = 1200
        size = 3
        mode = "reflect"
    
        images = th.generate_images(shape=(20, N, N))
    
        gpu_result = MedianFilter.filter_func(images.copy(), size, mode, force_cpu=False)
        cpu_result = MedianFilter.filter_func(images.copy(), size, mode, force_cpu=True)
    
>       npt.assert_almost_equal(gpu_result.data, cpu_result.data)
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 628825 / 28800000 (2.18%)
E       Max absolute difference: 0.8893112
E       Max relative difference: 28.829176
E        x: array([[[0.2275492, 0.6902805, 0.6902805, ..., 0.7283069, 0.6500515,
E                0.738842 ],
E               [0.3933151, 0.4027757, 0.3169855, ..., 0.6500515, 0.4720779,...
E        y: array([[[0.2275492, 0.6902805, 0.6902805, ..., 0.7283069, 0.6500515,
E                0.738842 ],
E               [0.3933151, 0.4027757, 0.3169855, ..., 0.6500515, 0.4720779,...

mantidimaging/core/gpu/test/gpu_test.py:75: AssertionError

```

### Context

Fail on Ubuntu 20.04, kernel 5.4.0-64-generic. Quadro T2000
`NVIDIA-SMI 460.32.03    Driver Version: 460.32.03    CUDA Version: 11.2 `

Passes on VM with Ubuntu 18.04, kernel 4.15.0, Quadro P4000,
`NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2`

### Failure Logs
See above

--- a/docs/release_notes/next.rst
+++ b/docs/release_notes/next.rst
@@ -12,3 +12,4 @@ Fixes
 
 - #816 : Error finding screen size in some dual screen set ups
 - #820 : Help links go to wrong page
+- #836 : Median filter returning data from wrong images

--- a/mantidimaging/core/gpu/utility.py
+++ b/mantidimaging/core/gpu/utility.py
@@ -129,6 +129,8 @@ def _send_arrays_to_gpu_with_pinned_memory(cpu_arrays, streams):
 
         for i in range(len(cpu_arrays)):
             gpu_arrays.append(_send_single_array_to_gpu(cpu_arrays[i], streams[i]))
+            # Synchronise to ensure that the upload has completed
+            streams[i].synchronize()
 
         return gpu_arrays
 

